### PR TITLE
Enable integration with cert-manager

### PR DESCRIPTION
**Pull request by Chen-Xintong**

### CODE REVIEW

File: deployment/integ-cert-manager/Integ-CM.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+kind: ClusterIssuer
+metadata:
+  name: istio-system
+spec:
+  ca:
+    secretName: istio-ca-selfsigned
+EOF
+$ kubectl apply -f ./istio-cm-issuer.yaml
+
+# Get the issuer certificate and replace cert in the chart
```
Comment (gyohuangxin) [inline deployment/integ-cert-manager/Integ-CM.md:38]: Can you give an example in the doc about how and where to replace cert in the Istio chart?

File: README.md [hunk 2]
```diff
@@ -16,13 +16,14 @@ The SDS Server can protect the private keys via SGX in 2 scenarios: workloads an
...
 - [Intel® SGX device plugin for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md)
 - Linux kernel version 5.11 or later on the host (in tree SGX driver)
 - [trusted-certificate-issuer](https://github.com/intel/trusted-certificate-issuer)
 - [Intel® SGX AESM daemon](https://github.com/intel/linux-sgx#install-the-intelr-sgx-psw)
 - [Intel® KMRA service](https://www.intel.com/content/www/us/en/developer/topic-technology/open/key-management-reference-application/overview.html)
-> NOTE: The KMRA service and AESM daemon is optional, needs to be set up only when remote attestaion required, which can be set through `NEED_QUOTE` flag in the chart.
+- [Intel® Linux SGX](https://github.com/intel/linux-sgx) and [cripto-api-toolkit](https://github.com/intel/crypto-api-toolkit) in the host (optional, only needed if you want to build sds-server image locally)
+> NOTE: The KMRA service and AESM daemon is also optional, needs to be set up only when remote attestaion required, which can be set through `NEED_QUOTE` flag in the chart.
```
Comment (gyohuangxin) [inline README.md:26]:
```suggestion
> NOTE: The KMRA service and AESM daemon are also optional, need to be set up only when remote attestation required, which can be set through `NEED_QUOTE` flag in the chart.
```

File: README.md [hunk 3]
```diff
@@ -16,13 +16,14 @@ The SDS Server can protect the private keys via SGX in 2 scenarios: workloads an
 
 Prerequisites for using Istio mTLS private key protection with SGX:
 
-- Kubernetes cluster with one or more nodes with Intel® [SGX](https://software.intel.com/content/www/us/en/develop/topics/software-guard-extensions.html) supported hardware
+- Kubernetes cluster with one or more nodes with [Intel® SGX](https://software.intel.com/content/www/us/en/develop/topics/software-guard-extensions.html) supported hardware
 - [Intel® SGX device plugin for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md)
 - Linux kernel version 5.11 or later on the host (in tree SGX driver)
 - [trusted-certificate-issuer](https://github.com/intel/trusted-certificate-issuer)
```
Comment (zhlsunshine) [inline README.md:22]: How about to change like this
```suggestion
- [trusted-certificate-issuer](https://github.com/intel/trusted-certificate-issuer) (Optional)
```

File: README.md [hunk 4]
```diff
@@ -16,13 +16,14 @@ The SDS Server can protect the private keys via SGX in 2 scenarios: workloads an
...
 - [Intel® SGX AESM daemon](https://github.com/intel/linux-sgx#install-the-intelr-sgx-psw)
```
Comment (zhlsunshine) [inline README.md:23]: How about to change like this

```suggestion
- [Intel® SGX AESM daemon](https://github.com/intel/linux-sgx#install-the-intelr-sgx-psw)  (Optional)
```

File: README.md [hunk 5]
```diff
@@ -23,7 +23,7 @@ Prerequisites for using Istio mTLS private key protection with SGX:
...
 - [Intel® KMRA service](https://www.intel.com/content/www/us/en/developer/topic-technology/open/key-management-reference-application/overview.html)
```
Comment (zhlsunshine) [inline README.md:24]: How about to change like this
```suggestion
- [Intel® KMRA service](https://www.intel.com/content/www/us/en/developer/topic-technology/open/key-management-reference-application/overview.html) (Optional)
```

File: README.md (hunk 3 above)
Comment (Chen-Xintong) [reply README.md]: Okay, fixed. I have mentioned in the `NOTE` below that the `KMRA` and `aesm` service are optional.

File: deployment/istio-configs/httpbin-hsm.yaml [hunk 6]
```diff
@@ -24,7 +24,7 @@ rules:
   resources: ["certificatesigningrequests", "certificatesigningrequests/approval", "certificatesigningrequests/status"]
   verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
 - apiGroups: ["certificates.k8s.io"]
-  resourceNames: [tcsissuer.tcs.intel.com/*]
+  resourceNames: ["clusterissuers.cert-manager.io/*", "tcsclusterissuer.tcs.intel.com/*", "tcsissuer.tcs.intel.com/*"]
```
Comment (zhlsunshine) [inline deployment/istio-configs/httpbin-hsm.yaml:27]: I think we should go through mTLS workload and gateway use cases on cert-manager and tcs before this configuration change.

File: deployment/istio-configs/sleep-hsm.yaml [hunk 7]
```diff
@@ -25,7 +25,7 @@ rules:
   resources: ["certificatesigningrequests", "certificatesigningrequests/approval", "certificatesigningrequests/status"]
   verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
 - apiGroups: ["certificates.k8s.io"]
-  resourceNames: [tcsclusterissuer.tcs.intel.com/*]
+  resourceNames: ["clusterissuers.cert-manager.io/*", "tcsclusterissuer.tcs.intel.com/*", "tcsissuer.tcs.intel.com/*"]
```
Comment (zhlsunshine) [inline deployment/istio-configs/sleep-hsm.yaml:28]: I think we should go through mTLS workload and gateway use cases on cert-manager and tcs before this configuration change.

**Review by zhlsunshine — APPROVED** [review]
LGTM

**Review by gyohuangxin — APPROVED** [review]

File: deployment/istio-configs/sleep-hsm.yaml (hunk 7 above)
Comment (Chen-Xintong) [reply deployment/istio-configs/sleep-hsm.yaml:28]: This yaml can work with cert manager, I'll double check integration with TCS today.

**Pull request merged**